spyder/plugins/pylint/widgets/pylintgui.py

```diff
@@ -44,6 +46,31 @@
     import gettext
     _ = gettext.gettext
 
+
+def _find_pylintrc_path(path):

```

Comment:
I suppose it could be, but IMO it makes much more sense to keep it somewhere inside the pylint plugin (maybe a different file if you suggest such), since its intrinsically part of the pylint plugin's implementation, not an external/general use utility. Furthermore, it would introduce complications, confusion and extra work for testing since it would either bifurcate the test code testing essentially the same thing at two different levels of abstraction, or require testing an external utility inside a plugin's tests.

Reply:
Yep, when I said a `utils` module I was thinking in a new file inside the `pylint` directory. Maybe following a directory structure like:
```
spyder/plugins/pylint/utils/path.py
```

Reply:
Oh, sorry, that makes sense then. I would suggest `spyder/plugins/pylint/utils/pylintrc.py` if design, spec or policy prohibits making it just a `utils` module in the base `pylint` dir. Given the relatively short length and unlikelihood of much else being added about path/pylintrc specifically as opposed to the higher possibility of more general pylint utils, for what its worth I would think it best to avoid adding yet another level of nesting to an already very deeply nested python package (given PEP 8 and `import this` (the Zen of Python) both strongly urge for keeping the namespace as flat as practical). However, this may be an consequence of the Spyder plugin spec or similar.

@ccordoba12 What do you suggest we go with?

Reply:
@goanpeca You may want to chime in too.

Reply:
I would do something like: 

`spyder/plugins/pylint/utils.py`

Reply:
Yeah, that was my preference too as I mentioned, I just wasn't sure if the Spyder plugin spec or policy/convention prohibited it based on @dalthviz not suggesting it himself. I'll implement it now.

Reply:
Implemented and pushed now, plus some more minor formatting cleanup and a few more code comments.